src/Workspaces/Core/Portable/Utilities/Documentation/XmlDocumentationProvider.cs

```diff
@@ -70,13 +77,9 @@ public ContentBasedXmlDocumentationProvider(byte[] xmlDocCommentBytes)
                 _xmlDocCommentBytes = xmlDocCommentBytes;
             }
 
-            protected override XDocument GetXDocument()
+            protected override Stream GetSourceStream()
             {
-                using (var stream = SerializableBytes.CreateReadableStream(_xmlDocCommentBytes, CancellationToken.None))
-                using (var xmlReader = XmlReader.Create(stream, s_xmlSettings))
-                {
-                    return XDocument.Load(xmlReader);
-                }
+                return SerializableBytes.CreateReadableStream(_xmlDocCommentBytes, CancellationToken.None);
```

Comment:
Nit: it looks like we have a cancellation token coming in. Should we pass it through?